test/unit/BPool/BPool_JoinPool.tree

```diff
@@ -0,0 +1,23 @@
+BPool::JoinPool
+├── when reentrancy lock is set
+│   └── it should revert
+├── when pool is not finalized
+│   └── it should revert
+├── when total supply is zero
+│   └── it should revert
+├── when pool amount out is zero
+│   └── it should revert
```

Comment:
is "too small"? i think it can revert with `poolAmountOut = totalSupply / 2 - 1`

Reply:
ah, is addressed in the next one, hmmm as you feel fit, i'd merge them both